src/renderers/shared/utils/Transaction.js

```diff
@@ -229,6 +236,8 @@ var Mixin = {
   },
 };
 
+export type TransactionType = typeof Transaction.Mixin;
```

Comment:
Would it make sense to just rename this to `Transaction` and then just do `module.exports = {` below? I always found the variable and then immediately reexporting pattern annoying anyway.

Besides, when we switch to ES modules the default export won't be an object anyway. Probably just named exports like:

```js
export Mixin;
export let OBSERVED_ERROR = {};
```

In Fiber everything is a type with the name of the module and an explicit create function.

https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactFiberReconciler.js#L54-L63

https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactFiber.js#L59-L111

https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactPriorityLevel.js#L15-L24 (This should be named exports too.)